metric_namer_test.go

```diff
@@ -21,259 +21,1019 @@ package otlptranslator
+		name     string
+		namer    MetricNamer
+		metric   Metric
+		expected string
 	}{
-		{"", "", ""},
-		{"s", "seconds", ""},
-		{"By/s", "bytes", "per_second"},
-		{"requests/m", "requests", "per_minute"},
-		{"{invalid}/second", "", "per_second"},
-		{"bytes/{invalid}", "bytes", ""},
-	}
+		// UTF8Allowed = false, WithMetricSuffixes = false tests
+		{
+			name: "simple metric name without suffixes",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: false,
+			},
+			metric: Metric{
+				Name: "simple_metric",
+				Unit: "",
+				Type: MetricTypeGauge,
+			},
+			expected: "simple_metric",
+		},
+		{
+			name: "metric with special characters replaced",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: false,
+			},
+			metric: Metric{
+				Name: "metric@with#special$chars",
+				Unit: "",
+				Type: MetricTypeGauge,
+			},
+			expected: "metric_with_special_chars",
+		},
+		{
+			name: "metric starting with digit gets underscore prefix",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: false,
+			},
+			metric: Metric{
+				Name: "123metric",
+				Unit: "",
+				Type: MetricTypeGauge,
+			},
+			expected: "_123metric",
+		},
+		{
+			name: "metric with namespace without suffixes",
+			namer: MetricNamer{
+				Namespace:          "test_namespace",
+				UTF8Allowed:        false,
+				WithMetricSuffixes: false,
+			},
+			metric: Metric{
+				Name: "simple_metric",
+				Unit: "",
+				Type: MetricTypeGauge,
+			},
+			expected: "test_namespace_simple_metric",
+		},
+		{
+			name: "empty metric name without suffixes",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: false,
+			},
+			metric: Metric{
+				Name: "",
+				Unit: "",
+				Type: MetricTypeGauge,
+			},
+			expected: "",
+		},
+		{
+			name: "metric with multiple consecutive special chars",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: false,
+			},
+			metric: Metric{
+				Name: "metric@@##$$name",
+				Unit: "",
+				Type: MetricTypeGauge,
+			},
+			expected: "metric_name",
+		},
+		{
+			name: "metric name with only special characters",
+			namer: MetricNamer{
+				UTF8Allowed:        false,
+				WithMetricSuffixes: false,
+			},
+			metric: Metric{
+				Name: "@#$%",
+				Unit: "",
+				Type: MetricTypeGauge,
+			},
+			expected: "",
+		},
 
-	for _, test := range tests {
-		mainUnitSuffix, perUnitSuffix := buildUnitSuffixes(test.unit)
-		require.Equal(t, test.expectedMain, mainUnitSuffix)
-		require.Equal(t, test.expectedPer, perUnitSuffix)
-	}
-}
+		{
+			name: "namespace with special characters",
+			namer: MetricNamer{
+				Namespace:          "test@namespace",
+				UTF8Allowed:        false,
+				WithMetricSuffixes: false,
+			},
+			metric: Metric{
+				Name: "metric",
+				Unit: "",
+				Type: MetricTypeGauge,
+			},
+			expected: "test@namespace_metric", // TODO: should be "test_namespace_metric"
```

Comment:
I did notice this bug while refactoring our tests!

Namespace is not being normalized even with UTF8Allowed = false